projects/organization-management/src/app/store/users/users.effects.ts

```diff
@@ -134,29 +137,58 @@ export class UsersEffects {
     )
   );
 
-  successMessageAfterUpdate$ = createEffect(() =>
+  updateCurrentUserRoles$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(updateUserSuccess, setUserRolesSuccess),
-      withLatestFrom(this.store.pipe(select(getSelectedUser), whenTruthy())),
-      map(([, user]) =>
-        displaySuccessMessage({
-          message: 'account.organization.user_management.update_user.confirmation',
-          messageParams: { 0: `${user.firstName} ${user.lastName}` },
-        })
+      ofType(setUserRolesSuccess),
+      mapToPayloadProperty('login'),
+      withLatestFrom(this.store.pipe(select(getLoggedInUser))),
+      filter(([login, currentUser]) => login === currentUser.login),
+      mapTo(loadRolesAndPermissions())
+    )
+  );
+
+  setUserBudgets$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(setUserBudgets),
+      mapToPayload(),
+      mergeMap(({ login, budgets }) =>
+        this.usersService.setUserBudgets(login, budgets).pipe(
+          withLatestFrom(this.store.pipe(select(selectPath))),
+          tap(([, path]) => {
+            if (path?.endsWith('users/:B2BCustomerLogin/budget')) {
+              this.navigateTo('../../' + login);
+            }
+          }),
+          map(([newBudgets]) => setUserBudgetsSuccess({ login, budgets: newBudgets })),
+          mapErrorToAction(setUserBudgetsFail, { login })
+        )
       )
     )
   );
 
-  updateCurrentUserRoles$ = createEffect(() =>
+  updateCurrentUserBudgets$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(setUserRolesSuccess),
+      ofType(setUserBudgetsSuccess),
       mapToPayloadProperty('login'),
       withLatestFrom(this.store.pipe(select(getLoggedInUser))),
       filter(([login, currentUser]) => login === currentUser.login),
       mapTo(loadRolesAndPermissions())
     )
   );
 
```

Comment:
Why do would need that? The other `updateCurrentUserRoles$` was there for updating the My Account User info label if the current user updates his own permissions, and we needed that because the current info is handled by the core PWA state. I don't see why we would reload roles and permissions for the current user if his budgets change. 🤔